### PR TITLE
Enable golangci-lint godot linter, fix errors

### PR DESCRIPTION
**Pull request by dfarrell07**
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

File: pkg/apis/submarinerconfig/v1alpha1/register.go [hunk 1]
```diff
@@ -10,19 +10,19 @@ var (
...
 	GroupVersion  = schema.GroupVersion{Group: GroupName, Version: "v1alpha1"}
 	schemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
-	// Install is a function which adds this version to a scheme
+	// Install is a function which adds this version to a scheme.
 	Install = schemeBuilder.AddToScheme
 
-	// SchemeGroupVersion generated code relies on this name
-	// Deprecated
+	// SchemeGroupVersion generated code relies on this name.
+	// DEPRECATED!
```
Comment (tpantelis) [inline pkg/apis/submarinerconfig/v1alpha1/register.go:17]: It's [recommended](https://go.dev/blog/godoc) that deprecated code begin with "Deprecated:" followed by some information about the deprecation.

```suggestion
	// Deprecated: generated code relies on SchemeGroupVersion.
```

File: pkg/apis/submarinerconfig/v1alpha1/register.go [hunk 2]
```diff
@@ -10,19 +10,19 @@ var (
...
 	SchemeGroupVersion = GroupVersion
-	// AddToScheme exists solely to keep the old generators creating valid code
-	// DEPRECATED
+	// AddToScheme exists solely to keep the old generators creating valid code.
+	// DEPRECATED!
```
Comment (tpantelis) [inline pkg/apis/submarinerconfig/v1alpha1/register.go:20]:
```suggestion
	// Deprecated: AddToScheme exists solely to keep the old generators creating valid code.
```

File: pkg/apis/submarinerconfig/v1alpha1/register.go [hunk 3]
```diff
@@ -10,19 +10,19 @@ var (
...
 	AddToScheme = schemeBuilder.AddToScheme
 )
 
 // Resource generated code relies on this being here, but it logically belongs to the group
-// DEPRECATED
+// DEPRECATED!
```
Comment (tpantelis) [inline pkg/apis/submarinerconfig/v1alpha1/register.go:25]:
```suggestion
// Deprecated: generated code relies on Resource being present, but it logically belongs to the group.
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/apis/submarinerconfig/v1alpha1/register.go (hunk 1 above)
Comment (dfarrell07) [reply pkg/apis/submarinerconfig/v1alpha1/register.go:17]: Yep, that seems a lot more useful than an exclamation mark. :wink:

**Pull request merged**